gateway/handlers/notifiers.go

```diff
@@ -46,19 +46,26 @@ type PrometheusFunctionNotifier struct {
 }
 
 // Notify records metrics in Prometheus
-func (p PrometheusFunctionNotifier) Notify(method string, URL string, originalURL string, statusCode int, duration time.Duration) {
-	seconds := duration.Seconds()
-	serviceName := getServiceName(originalURL)
+func (p PrometheusFunctionNotifier) Notify(method string, URL string, originalURL string, statusCode int, event string, duration time.Duration) {
```

Comment:
Can we use an enum for the `event` instead of an arbitrary string? it will be more type safe, easier to read in the docs and IDEs, and we can avoid bugs to do typos.  Especially if we pair it with switch below on the enum constants.

```go
type FunctionEvent int
const (
   FunctionStarted = iota
   FunctionEnded
)
func (d FunctionEvent) String() string {
    return [...]string{"started", "ended"}[d]
}
```

Reply:
I don't disagree. This is an experimental PR to see if we can solve at least one of the issues for the user who is in production experiencing this problem today

Reply:
¯\_(ツ)_/¯ if you are happy with it, then i guess we can leave it

Reply:
Seems like a reasonable enhancement